### PR TITLE
Laboratory work #1, Fedorov Makar - 19FPL1

### PR DESCRIPTION
**Pull request by makar-fedorov**

### CODE REVIEW

**Comment by WhiteJaeger** [comment]
Посмотрите, пожалуйста, видео, которое мы вам отправили вчера (14.9) на почту группы - там есть инструкция по созданию PR и их оформлению.

**Comment by ValeriyaKuznetsova** [comment]
Unfortunately, the deadline is not that far. Please, push changes so that we can review them.

File: lab_1/main.py [hunk 1]
```diff
@@ -13,7 +13,16 @@ def tokenize(text: str) -> list:
...
-    pass
+    if type (text) == str:
+        clean_text = ''
+        for sign in text:
+            if sign.lower() in 'abcdefghijklmnopqrstuvwxyz ':
+                clean_text += sign.lower()
+        tokens = clean_text.split()
+        return (tokens)
+    else:
+        return ([])
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: why ([])? you can remove brackets

File: lab_1/main.py [hunk 2]
```diff
@@ -13,7 +13,16 @@ def tokenize(text: str) -> list:
...
     """
-    pass
+    if type (text) == str:
+        clean_text = ''
+        for sign in text:
+            if sign.lower() in 'abcdefghijklmnopqrstuvwxyz ':
+                clean_text += sign.lower()
+        tokens = clean_text.split()
+        return (tokens)
+    else:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: you can remove "else", if "return tokens" works, the function will stop its work anyway

File: lab_1/main.py [hunk 3]
```diff
@@ -13,7 +13,16 @@ def tokenize(text: str) -> list:
...
     --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
     """
-    pass
+    if type (text) == str:
+        clean_text = ''
+        for sign in text:
+            if sign.lower() in 'abcdefghijklmnopqrstuvwxyz ':
+                clean_text += sign.lower()
+        tokens = clean_text.split()
+        return (tokens)
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: remove brackets (everywhere)

File: lab_1/main.py [hunk 4]
```diff
@@ -26,19 +35,44 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
     stop_words = ['the', 'is']
     --> ['weather', 'sunny', 'man', 'happy']
     """
-    pass
-
+    if type (tokens) == list:
+        for token in tokens:
+            for sign in token:
+                if sign not in 'abcdefghijklmnopqrstuvwxyz':
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: We consider that tokens have already been "cleaned". You can remove this check.

File: lab_1/main.py [hunk 5]
```diff
@@ -26,19 +35,44 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
+                    tokens.clear()
+                    break
+        if type (stop_words) == list:
+            for word in stop_words:
+                for token in tokens:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: Can you suggest a way to do that with only one "for"?

File: lab_1/main.py [hunk 6]
```diff
@@ -26,19 +35,44 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
+                    if word == token:
+                        tokens.remove(token)
+            return(tokens)
+        else:
+            return (tokens)
+    else:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: remove one "else: return tokens"

File: lab_1/main.py [hunk 7]
```diff
@@ -26,19 +35,44 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
+        return ([])
+#remove_stop_words ()
 
 def calculate_frequencies(tokens: list) -> dict:
     """
-    Calculates frequencies of given tokens
+    Calculates frequencies of given tokens{'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: this dictionary doesn't belong here

File: lab_1/main.py [hunk 8]
```diff
@@ -26,19 +35,44 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
     :return: a dictionary with frequencies
     e.g. tokens = ['weather', 'sunny', 'man', 'happy']
     --> {'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
     """
-    pass
-
+    if type (tokens) == list:
+        for token in tokens:
+            for sign in token:
+                if sign not in 'abcdefghijklmnopqrstuvwxyz':
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: same, remove this check

File: lab_1/main.py [hunk 9]
```diff
@@ -26,19 +35,44 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
+                    tokens.clear()
+                    break
+        token_frequency = {token: tokens.count(token) for token in tokens}
+        print (token_frequency)
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: do not forget to remove prints when you push something

File: lab_1/main.py [hunk 10]
```diff
@@ -50,8 +84,38 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list:
     top_n = 1
     --> ['happy']
     """
-    pass
-
+    if type (freq_dict) == dict and type (top_n) == int:
+        for key in freq_dict.keys():
+            for sign in key:
+                if sign not in 'abcdefghijklmnopqrstuvwxyz':
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: remove

File: lab_1/main.py [hunk 11]
```diff
@@ -50,8 +84,38 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list:
...
+                    return ([])
+        for value in freq_dict.values():
+            if type (value) != int:
+                return ([])
+        spisok_values = []
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: list_of_values or just values?
please, rename these variables

File: lab_1/main.py [hunk 12]
```diff
@@ -70,7 +134,42 @@ def get_concordance(tokens: list, word: str, left_context_size: int, right_conte
...
+    if type (tokens) == list and type (word) == str:
+        for token in tokens:
+            for sign in token:
+                if sign not in 'abcdefghijklmnopqrstuvwxyz':
+                    tokens.clear()
+                    return ([])
+        for sign in word:
+            if sign not in 'abcdefghijklmnopqrstuvwxyz':
+                return ([])
+        if type (left_context_size) == int and type (right_context_size) == int:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: put this check at the beginning of the function

File: lab_1/main.py [hunk 13]
```diff
@@ -70,7 +134,42 @@ def get_concordance(tokens: list, word: str, left_context_size: int, right_conte
...
+                    if left_context_size < 1 and right_context_size < 1:
+                        return ([])
+                    current_context_list = left_context
+                    current_context_list.append(token)
+                    current_context_list.extend (right_context)
+                    concordance.append(current_context_list)
+                    tokens.insert(tokens.index(token), 'буферный элемент')
+                    tokens.remove(token)
+            return (concordance)
+        else:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: remove one else:return

File: lab_1/main.py [hunk 14]
```diff
@@ -88,7 +187,29 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
     right_n = 3
     --> [['man', 'is'], ['dog, 'cat']]
     """
-    pass
+    adjacent_words = []
+    concordance = get_concordance (tokens, word, left_n, right_n)
+    if type(left_n) == int and type(right_n) == int:
+        for spisok in concordance:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: please, rename

File: lab_1/main.py [hunk 15]
```diff
@@ -106,7 +227,12 @@ def write_to_file(path_to_file: str, content: list):
     """
     Writes the result in a file
     """
-    pass
+    file = open (path_to_file, 'w')
+    for spisok in content:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: rename

File: lab_1/main.py [hunk 16]
```diff
@@ -106,7 +227,12 @@ def write_to_file(path_to_file: str, content: list):
...
+        text = str(spisok) + '\n'
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: The expected result:
the weather
sunny the man
happy the dog
...

**Comment by ValeriyaKuznetsova** [comment]
Can you push the changes, please?

**Comment by ValeriyaKuznetsova** [comment]
do not forget to put "Review required" label

File: lab_1/main.py [hunk 17]
```diff
@@ -25,7 +39,14 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
     stop_words = ['the', 'is']
     --> ['weather', 'sunny', 'man', 'happy']
     """
-    pass
+    new_tokens = []
+    if type (tokens) == list:
+        if type (stop_words) == list:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: join lines 43, 44

File: lab_1/main.py [hunk 18]
```diff
@@ -105,7 +224,14 @@ def write_to_file(path_to_file: str, content: list):
...
     Writes the result in a file
     """
-    pass
+    text = ''
+    file = open (path_to_file, 'w')
+    for small_list in content:
+        for word in small_list:
+            text += word + ' '
+        text += '\n'
+        file.write (text)
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: put all joined lines into a variable
only then write it to file

it is not good to write to file at each iteration

File: lab_1/start.py [hunk 19]
```diff
@@ -1,18 +1,23 @@
...
     current_dir = os.path.dirname(os.path.abspath(__file__))
-    data = read_from_file(os.path.join(current_dir, 'data.txt'))
+    data = read_from_file(os.path.join(current_dir, 'lab_1/data.txt'))
     stop_words = []
 
     #  here goes your logic: calling methods from concordance.py
-
-    RESULT = None
+    text = read_from_file('lab_1/data.txt')
+    tokens = tokenize(text)
```
Comment (ValeriyaKuznetsova) [inline lab_1/start.py]: can you add all other functions?

**Review by ValeriyaKuznetsova — COMMENTED** [review]
correct these issues (and previous one - remove prints)
work on tests and lint

File: lab_1/main.py [hunk 20]
```diff
@@ -172,50 +163,47 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
     """
     adjacent_words = []
     concordance = get_concordance (tokens, word, left_n, right_n)
-    if isinstance(tokens, list) and isinstance(word, str) and isinstance(left_n, int) and isinstance(right_n, int):
+    if (isinstance(tokens, list) and isinstance(word, str) and isinstance(left_n, int) and isinstance(right_n, int)
+            and not isinstance (left_n, bool) and not isinstance (right_n, bool)):
+        def aid_function (tokens, word, left_n, right_n):
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: no defs into def
remove it

File: lab_1/sort_concordance_test.py [hunk 21]
```diff
@@ -123,7 +123,7 @@ def test_get_concordance_several_contexts_big_text_left(self):
         Checks if contexts for a given term can be found in real text properly
         Taking into consideration left context
         """
-        text = read_from_file('lab_1/data.txt')
+        text = read_from_file('data.txt')
```
Comment (ValeriyaKuznetsova) [inline lab_1/sort_concordance_test.py]: don't push such changes
change back everywhere

**Comment by ValeriyaKuznetsova** [comment]
in sort_concordance_test.py change back a line:
 text = read_from_file('lab_1/data.txt')

File: lab_1/main.py [hunk 22]
```diff
@@ -121,29 +121,31 @@ def get_concordance(tokens: list, word: str, left_context_size: int, right_conte
...
-                    return []
-                current_context_list = left_context
-                current_context_list.append(token)
-                current_context_list.extend (right_context)
-                concordance.append(current_context_list)
-                tokens.insert(tokens.index(token), 'буферный элемент')
-                tokens.remove(token)
-        return concordance
-    return []
+    if not (isinstance(left_context_size, int) and isinstance (right_context_size, int)):
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: not isinstance and not isinst

File: lab_1/main.py [hunk 23]
```diff
@@ -121,29 +121,31 @@ def get_concordance(tokens: list, word: str, left_context_size: int, right_conte
...
+        if not (isinstance (tokens, list) and isinstance (word, str) and not isinstance (left_context_size, bool)
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: not and not and (NO not) and (NO not)

File: lab_1/main.py [hunk 24]
```diff
@@ -163,38 +165,37 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
...
-            elif left_n < 1 and right_n < 1:
-                return []
-            adjacent_words.append(current_words)
-        for small_list in concordance:
-            for token in small_list:
-                if token == word:
-                    aid_function(tokens, word, left_n, right_n)
-        return adjacent_words
-    return []
+    if not (isinstance(tokens, list) and isinstance(word, str) and isinstance(left_n, int)
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: same

File: lab_1/start.py [hunk 25]
```diff
@@ -20,14 +20,15 @@
...
-    stop_words = read_from_file(os.path.join(current_dir, 'stop_words.txt'))
+    stop_words = read_from_file(os.path.join(current_dir, 'stop_words.txt')).split()
     clean_text_list = remove_stop_words(tokens, stop_words)
     frequencies_dict = calculate_frequencies(tokens)
     top_n_words = get_top_n_words(frequencies_dict, 5)
     concordance = get_concordance(tokens, 'sodium', 2, 2)
     adjacent_words = get_adjacent_words(tokens, 'sodium', 2, 2)
 
     RESULT = sort_concordance(tokens, 'sodium', 1, 1, True)
+    print (RESULT)
```
Comment (ValeriyaKuznetsova) [inline lab_1/start.py]: do not forget to remove print

**Comment by ValeriyaKuznetsova** [comment]
Open another PR with the second lab

**Pull request closed without merging**